### PR TITLE
Create measurement screen

### PR DESCRIPTION
**Pull request by iOSKirill**
Added a black pulse measurement option

### CODE REVIEW

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        let specs = VideoSpec(fps: 30)
+        heartRateManager = HeartRateManager(cameraType: .back, preferredSpec: specs)
+        heartRateManager!.imageBufferHandler = { [unowned self] imageBuffer in
+            self.handle(buffer: imageBuffer)
+        }
+    }
+
+    // MARK: - AVCaptureSession Helpers
+    func initCaptureSession() {
+        heartRateManager!.startCapture()
```
Comment (ghost) [inline HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift]: please don't use force unwrap

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    private var inputs: [CGFloat] = []
+    private var measurementStartedFlag = false
+    private var timer = Timer()
+    @Published var pulse: String = ""
+
+    // MARK: - Frames Capture Methods
+    func initVideoCapture() {
+        let specs = VideoSpec(fps: 30)
+        heartRateManager = HeartRateManager(cameraType: .back, preferredSpec: specs)
+        heartRateManager!.imageBufferHandler = { [unowned self] imageBuffer in
```
Comment (ghost) [inline HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift]: please don't use force unwrap

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        }
+    }
+
+    // MARK: - AVCaptureSession Helpers
+    func initCaptureSession() {
+        heartRateManager!.startCapture()
+    }
+
+    func deinitCaptureSession() {
+        heartRateManager!.stopCapture()
```
Comment (ghost) [inline HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift]: same here

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    // MARK: - Measurement
+    private func startMeasurement() {
+        DispatchQueue.main.async {
+            self.toggleTorch(status: true)
+            self.timer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true, block: { [weak self] _ in
+                guard let self = self else { return }
+                let average = self.pulseDetector.getAverage()
+                let pulse = 60.0 / average
+                if pulse == -60 {
+                    self.pulse = "Expect the pulse to be measured..."
```
Comment (ghost) [inline HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift:57]: please localize

File: HeartRateMonitorApp/HeartRateMonitorApp/Services/HeartRateManager.swift [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+struct VideoSpec {
+    var fps: Int32?
+    var size: CGSize?
+}
+
+typealias ImageBufferHandler = ((_ imageBuffer: CMSampleBuffer) -> Void)
+
+class HeartRateManager: NSObject {
```
Comment (ghost) [inline HeartRateMonitorApp/HeartRateMonitorApp/Services/HeartRateManager.swift]: final?

Comment (iOSKirill) [reply HeartRateMonitorApp/HeartRateMonitorApp/Services/HeartRateManager.swift]: Fixed

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift (hunk 3 above)
Comment (iOSKirill) [reply HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift]: Fixed

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift (hunk 1 above)
Comment (iOSKirill) [reply HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift]: Fixed

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift (hunk 2 above)
Comment (iOSKirill) [reply HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift]: Fixed

File: HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift (hunk 4 above)
Comment (iOSKirill) [reply HeartRateMonitorApp/HeartRateMonitorApp/MeasurementModule/ViewModel/MeasurementViewModel.swift:57]: This is just a stub to check if the measurement logic is working

**Review by ghost — APPROVED** [review]

**Pull request merged**